docs/scripts/check_links.py

```diff
@@ -0,0 +1,66 @@
+"""Utility for checking the links of the documentation."""
+
+import argparse
+import os
+from subprocess import check_call, run
+
+from baybe.telemetry import VARNAME_TELEMETRY_ENABLED
+
+parser = argparse.ArgumentParser()
+parser.add_argument(
+    "-e",
+    "--ignore_examples",
+    help="Ignore the examples by not executing them.",
+    action="store_true",
+)
+parser.add_argument(
+    "-w",
+    "--include_warnings",
+    help="Include warnings when processing the examples. The default is ignoring them.",
+    action="store_true",
+)
+parser.add_argument(
+    "-f",
+    "--force",
+    help="Force-build the documentation, even when there are warnings or errors.",
+    action="store_true",
+)
+
+# Parse input arguments
+args = parser.parse_args()
+IGNORE_EXAMPLES = args.ignore_examples
+INCLUDE_WARNINGS = args.include_warnings
+FORCE = args.force
+
+
+def check_links(force: bool = False) -> None:
+    """Check whether the links of the documentation are valid.
+
+    Args:
+        force: Force-check the links, even if there are errors or wanrings.
+    """
+    # The call for checking external links.
+    link_call = [
+        "sphinx-build",
+        "-b",
+        "linkcheck",
+        "docs",
+        "docs/build",
+    ]
+
+    if force:
+        print("Force-building the documentation, ignoring errors and warnings.")
```

Comment:
Perhaps I simply don't get it, but what is the point of force-executing the linkcheck? Isn't identifying failures exactly the purpose of the script? Or are there other types of "failures" (except broken links) that we want to bypass?

Reply:
Didn't we just recently have exactly that situation when we renamed one of the user guides? This made `linkcheck` fail, but we knew that this is fine, yet it required us to force-linkcheck.

Reply:
Yes, but the point here is that `tox -e docs` is called in CI, which includes the linkcheck that failed. Wouldn't it make more sense that, in case the `--force` option is given, we simply skip the linkcheck in the `docs` command instead of running it with `--force`, which I think fulfills no purpose at all, at least as far as I understand?

Reply:
Agree that linkcheck with force does not make a lot of sense. However,  this should really be an edge cases, and I would not adjust all of the code just to try to make an edge case be slightly faster in the CI. Also, since the force option is used as a positional argument, we would need to change this OR adjust the call in such a way that when linkcheck is called with the force option, it just immediately returns/terminates. Would this work for you?

Reply:
Yes, if this help to simplify the tox pipeline, I think you can do it that way 👍🏼 Highest priority is that no linkcheck is triggered in the first place

Reply:
Since the individual parts of the pipeline can now basically be toggled on and off, I removed the option to do the linkcheck. This raises the question if this file itself if still necessary, but do speed things up I would just keep it for now (might also be useful if we see that we should actually have a force option here).